core/src/main/java/org/apache/iceberg/inmemory/InMemoryCatalog.java

```diff
@@ -138,13 +138,13 @@ public boolean dropTable(TableIdentifier tableIdentifier, boolean purge) {
 
   @Override
   public List<TableIdentifier> listTables(Namespace namespace) {
-    if (!namespaceExists(namespace) && !namespace.isEmpty()) {
+    if (!namespaceExists(namespace)) {
```

Comment:
I don't think we want to change the behavior here. And if we did, how is this more correct?

I think it is fine for catalogs to consider this an independent namespace. And in this case, it makes sense that the root namespace always exists.

Reply:
I've removed this so that we can handle empty namespaces separately in `InMemoryCatalog`